### ISSUE TITLE
Unable to associate Execution Node with Hop Node after adding the 29th instance. "Something went wrong..."

### ISSUE THREAD

**Issue by cselbach** [ansible/awx#15698]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

I am experiencing a consistent issue when attempting to associate a Hop-Node under peers for an Execution Node in AWX. The problem occurs when there are 28 Execution Nodes. After adding a 29th instance, it leads to an error in the UI when I attempt to associate a Hop-Node for this object. This seems to be a reproducible bug.

Any help or guidance on resolving this issue would be greatly appreciated. Thank you!

### AWX version

24.6.1

### Select the relevant components

- [X] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Installation method

kubernetes

### Modifications

no

### Web browser

Chrome, Safari

### Steps to reproduce

- Set up AWX with 28 Execution Nodes and a Hop-Node (AWXMeshIngress deployed with crd in Kubernetes) .
- Add a 29th Execution Node.
- Attempt to associate the Hop-Node under peers for the new Execution Node.
- Observe the error in the UI.

### Expected results

The Hop-Node should be successfully associated under peers for the Execution Node without any errors.

### Actual results

An error message appears in the UI, and the association fails. The error message displayed is:

Something went wrong...
There was an error loading this content. TypeError: Cannot read properties of undefined (reading 'hostname')
TypeError: Cannot read properties of undefined (reading 'hostname')
    at p (....

![Bildschirmfoto 2024-12-10 um 18 23 32](https://github.com/user-attachments/assets/c7e453f1-9ec9-4891-a8f7-560e9c12b3ed)

### Additional information

This issue appears on both Google Chrome and Safari browsers (other browsers not tested). The issue appears to be related to the number of Execution Nodes.

**Comment by fosterseth** [comment]
are you able to associate the instance in the API?

**Comment by cselbach** [comment]
Yes, association via API is possible without any issues and without error messages. However, when you look at the UI, the error message still appears if a certain number of instances have been created. Functionally, association via the UI also works, but it does indeed seem to be an issue with the display in the UI.

**Comment by tanganellilore** [comment]
Hi,

same for me with 24.6.1  and 24.2.0 .

Everything via api works.


Thanks
